src/server/plugins/addSecurityHeaders.js

```diff
@@ -14,11 +14,39 @@
  * permissions and limitations under the License.
  */
 
-export default function addSecurityHeaders(req, res, next) {
-  res.set('X-Frame-Options', 'DENY');
-  res.set('X-Content-Type-Options', 'nosniff');
-  res.set('Strict-Transport-Security', 'max-age=15552000; includeSubDomains');
-  res.set('X-XSS-Protection', '1; mode=block');
-  res.set('Referrer-Policy', process.env.ONE_REFERRER_POLICY_OVERRIDE || 'same-origin');
-  next();
-}
+import fp from 'fastify-plugin';
+
+/**
+ * Fastify Plugin that adds security into headers
+ * @param {import('fastify').FastifyInstance} fastify Fastify instance
+ * @param {import('fastify').FastifyPluginOptions} _opts plugin options
+ * @param {import('fastify').FastifyPluginCallback} done plugin callback
+ */
+const addSecurityHeaders = (fastify, opts = {}, done) => {
+  const ignoreRoutes = opts.ignoreRoutes || [];
+
+  fastify.addHook('onRequest', async (request, reply) => {
+    reply.header('Strict-Transport-Security', 'max-age=15552000; includeSubDomains');
+    reply.header('x-dns-prefetch-control', 'off');
```

Comment:
Why does this include so many headers that were not there previously?

Reply:
these are headers required by the integration tests that are not widely present. For the case of `x-dns-prefetch-control`, it is added by Helmet (see https://github.com/helmetjs/helmet/blob/a2549bd4688d9e711aea64a7fdffb07f4136f6cf/middlewares/x-dns-prefetch-control/index.ts#L17). Helmet, in the Fastify implementation, is only used on routes that will render Html, it isn't needed for all urls. By doing so, we increase performance. Right now, there's an integration test that expects this header in a route that doesn't render html (see https://github.com/americanexpress/one-app/blob/822999488c237a66298652d5746af65400234fd4/__tests__/integration/one-app.spec.js#L1365)

if we feel that we do not actually need them for all routes them we can delete them, they were added to support backwards compatibility